lightning/src/ln/channel.rs

```diff
@@ -2300,6 +2300,13 @@ impl<Signer: Sign> Channel<Signer> {
 		}, commitment_signed, closing_signed, monitor_update))
 	}
 
+	/// Public version of the below, checking relevant preconditions first
+	pub fn maybe_free_holding_cell_htlcs<L: Deref>(&mut self, logger: &L) -> Result<(Option<(msgs::CommitmentUpdate, ChannelMonitorUpdate)>, Vec<(HTLCSource, PaymentHash)>), ChannelError> where L::Target: Logger {
+		if self.channel_state >= ChannelState::ChannelFunded as u32 &&
+		   (self.channel_state & (ChannelState::AwaitingRemoteRevoke as u32 | ChannelState::PeerDisconnected as u32 | ChannelState::MonitorUpdateFailed as u32)) == 0 {
```

Comment:
Hmmm, I don't think so - on a graceful shutdown start we drop HTLC adds and wait until later for the HTLC fulfill/fails, but we still want to send those - ultimately we're waiting on those to finish so that we can do a graceful shutdown. Once we're "really" shut down (either forced or done negotiating the graceful shutdown), we drop the Channel object so we can't get here anyway.